src/Illuminate/Support/Number.php

```diff
@@ -17,6 +18,136 @@ class Number
 
+    /**
+     * Get a new numeral instance for the given value.
+     *
+     * @param  int|float  $value
+     */
+    public static function of($value): Numeral
+    {
+        return new Numeral($value);
+    }
+
+    /**
+     * Determine if the given value is a number.
+     *
+     * @param  mixed  $number
+     */
+    public static function isNumeric($number): bool
+    {
+        return is_numeric($number);
+    }
+
+    /**
+     * Determine if the given value is an even number.
+     *
+     * @param  mixed  $value
+     */
+    public static function isEven($value): bool
+    {
+        return $value % 2 === 0;
+    }
+
+    /**
+     * Determine if the given value is an odd number.
+     *
+     * @param  mixed  $value
+     */
+    public static function isOdd($value): bool
+    {
+        return $value % 2 !== 0;
+    }
+
+    /**
+     * Determine if the given value is a float.
+     *
+     * @param  mixed  $value
+     */
+    public static function isFloat($value): bool
+    {
+        return self::isNumeric($value) && is_float($value + 0);
+    }
+
+    /**
+     * Determine if the given value is an integer.
+     *
+     * @param  mixed  $value
+     */
+    public static function isInt($value): bool
+    {
+        return self::isNumeric($value) && is_int($value + 0);
+    }
+
+    /**
+     * Determine if the given value is a positive number.
+     *
+     * @param  mixed  $value
+     */
+    public static function isPositive($value): bool
+    {
+        return $value > 0;
+    }
+
+    /**
+     * Determine if the given value is a positive integer.
+     *
+     * @param  mixed  $value
+     */
+    public static function isPositiveInt($value): bool
+    {
+        return self::isInt($value) && self::isPositive($value);
+    }
+
+    /**
+     * Determine if the given value is a positive float.
+     *
+     * @param  mixed  $value
+     */
+    public static function isPositiveFloat($value): bool
+    {
+        return self::isFloat($value) && self::isPositive($value);
+    }
+
+    /**
+     * Determine if the given value is a negative number.
+     *
+     * @param  mixed  $value
+     */
+    public static function isNegative($value): bool
+    {
+        return $value < 0;
+    }
+
+    /**
+     * Determine if the given value is a negative integer.
+     *
+     * @param  mixed  $value
+     */
+    public static function isNegativeInt($value): bool
+    {
+        return self::isInt($value) && self::isNegative($value);
+    }
+
+    /**
+     * Determine if the given value is a negative float.
+     *
+     * @param  mixed  $value
+     */
+    public static function isNegativeFloat($value): bool
+    {
+        return self::isFloat($value) && self::isNegative($value);
+    }
+
+    /**
+     * Determine if the given value is zero.
+     *
+     * @param  mixed  $value
+     */
+    public static function isZero($value): bool
+    {
+        return (int) $value === 0;
+    }
```

Comment:
`0.5` is considered zero?

Reply:
Not an absolute zero, no, that will fall in the field of floats

However, thanks to this comment I just realized that this might lead to false positives, as the cast is applied to the result of the comparison, not the `$value` itself. I'll update it after this conversation has been resolved (just to preserve the comments)